src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs

```diff
@@ -198,8 +197,15 @@ public TimeSpan NonceLifetime
         /// <summary>
         /// Gets or sets a value indicating if a 'sub' claim is required.
         /// </summary>
-        [DefaultValue(false)]
-        public bool RequireSub { get; set; }
+        [DefaultValue(true)]
+        public bool RequireSub { get; set; } = RequireSubByDefault;
```

Comment:
@polita @ChenYan98 This may be breaking to some users. The specification requires a 'sub' claim, we relaxed that in our first versions as some IDP's were not including the sub claim. Our old design would have required users to derive from OpenIdConnectProtocolValidator to choose a different value.  This design, allows a simpler implementation just deploy with setting RequireSubByDefault = false.